eng/pipelines/templates/BuildAndTest.yml

```diff
@@ -38,8 +38,31 @@ steps:
               /bl:${{ parameters.repoLogPath }}/build.binlog
               $(_OfficialBuildIdArgs)
               $(_InternalBuildArgs)
+              /p:SkipTestProjects=true
       displayName: Build
 
+    - script: ${{ parameters.dotnetScript }}
+              build
+              tests/workloads.proj
+              -p:SkipPackageCheckForTemplatesTesting=true
+      displayName: Prepare sdks for templates testing
+
+    - script: ${{ parameters.buildScript }}
+              -build
+              -restore
```

Comment:
Why do we need to restore and build again? This can add anywhere between 40s and 3min...

Reply:
Since we are not doing a restore for all the test projects earlier, we need to do that for the templates test project here.